python/lolopy/learners.py

```diff
@@ -0,0 +1,204 @@
+            x_i_java = ListConverter().convert(x_i, self.gateway._gateway_client)
+            x_i_java = self.gateway.jvm.scala.collection.JavaConverters.asScalaBuffer(
+                x_i_java).toVector()
+            X_java.append(x_i_java)
+        X_java = self.gateway.jvm.scala.collection.JavaConverters.asScalaBuffer(X_java)
+        return X_java
+
+
+class BaseRandomForest(BaseLoloLearner):
+    """Random Forest """
+
+    def __init__(self, num_trees=-1, useJackknife=True, subsetStrategy=4):
+        """Initialize the RandomForest
+
+        Args:
+            num_trees (int): Number of trees to use in the forest
+        """
+        super(BaseRandomForest, self).__init__()
+
+        # Get JVM for this object
+
+        # Store the variables
+        self.num_trees = num_trees
+        self.useJackknife = useJackknife
+        self.subsetStrategy = subsetStrategy
+
+    def _make_learner(self):
+        """Instantiate the learner used by Lolo to train a model
+
+        Ret"""
+        #  TODO: Figure our a more succinct way of dealing with optional arguments/Option values
+        learner = self.gateway.jvm.io.citrine.lolo.learners.RandomForest(
+            self.num_trees, self.useJackknife,
+            getattr(self.gateway.jvm.io.citrine.lolo.learners.RandomForest,
+                    "$lessinit$greater$default$3")(),
+            getattr(self.gateway.jvm.io.citrine.lolo.learners.RandomForest,
+                    "$lessinit$greater$default$4")(),
+            self.subsetStrategy
+        )
+        return learner
+
+    def predict(self, X, return_std=False):
+        # Convert the data to Java
+        X_java = self._convert_run_data(X)
+
+        # Get the PredictionResult
+        pred_result = self.model_.transform(X_java)
+
+        # Pull out the expected values
+        exp_values = pred_result.getExpected()
+        y_pred = [exp_values.apply(i) for i in range(len(X))]
+
+        # If desired, return the uncertainty too
+        if return_std:
+            # TODO: This part fails on Windows because the NativeSystemBLAS is not found. Fix that
+            # TODO: This is only valid for regression models. Perhaps make a "LoloRegressor" class
+            uncertain = pred_result.getUncertainty().get()
+            y_std = np.zeros_like(y_pred)
+            for i in range(len(X)):
+                y_std[i] = uncertain.apply(i)
+            return y_pred, y_std
+
+        # Get the expected values
+        return y_pred
+
+
+class RandomForestRegressor(BaseRandomForest, RegressorMixin):
+    """Random Forest for regression"""
+    pass
+
+
+class RandomForestClassifier(BaseRandomForest, ClassifierMixin):
+
+    def fit(self, X, y, weights=None):
+        # Get the number of classes
+        self.n_classes_ = len(set(y))
+
+        return super(RandomForestClassifier, self).fit(X, y, weights)
+
+    def predict(self, X):
+        return super(RandomForestClassifier, self).predict(X, False)
+
+    def predict_proba(self, X):
```

Comment:
`predict_proba`?  Did you mean `predict_probability`?

Reply:
No, `predict_proba` is the proper function name for scikit-learn classifiers.